tests/modules/update_availabilities_fullfilled/app/update_availabilities_fullfilled_usecase.test.ts

```diff
@@ -0,0 +1,546 @@
+      },
+      {
+        userId: 3,
+        possibilityId: '113e4567-e89b-12d3-a456-426614174000',
+        classId: '0a8c5357-1f07-5b24-9845-9318c47ac924',
+      }
+    ]
+
+    const lengthUserAvsFullfilledBefore = await repo.getAvsFullfilledLength() 
+
+    await expect(usecase.execute(scheduleId, avsFullfilledParam)).rejects.toThrowError(
+      'The professor with id 4 cannot teach the class with id 0a8c5357-1f07-5b24-9845-9318c47ac926',
+    )
+
+    const lengthUserAvsFullfilledAfter = await repo.getAvsFullfilledLength()
+
+    expect(lengthUserAvsFullfilledBefore).toEqual(lengthUserAvsFullfilledAfter)
+  })
+  it('Should raise error: professor does not have availability for certain possibility', async () => {
+    const repo = new ScheduleRepositoryMock()
+    const usecase = new UpdateAvailabilitiesFullfilledUsecase(repo)
+    const scheduleId = "2S-4CM-D5@2024(SCS)"
+    const avsFullfilledParam : AvailabilitiesFullfilledParam[] = [
+      {
+        userId:3,
+        classId:"0a8c5357-1f07-5b24-9845-9318c47ac926",
+        possibilityId:"124e4567-e89b-12d3-a456-426614174000"
+      },
+      {
+        userId:4,
+        possibilityId:"143e4567-e89b-12d3-a456-426614174000",
+        classId:"0a8c5357-1f07-5b24-9845-9318c47ac927"
+      },
+      {
+        userId: 3,
+        possibilityId: '113e4567-e89b-12d3-a456-426614174000',
+        classId: '0a8c5357-1f07-5b24-9845-9318c47ac924',
+      }
+    ]
+
+    const lengthUserAvsFullfilledBefore = await repo.getAvsFullfilledLength() 
+
+    await expect(usecase.execute(scheduleId, avsFullfilledParam)).rejects.toThrowError(
+      "The professor with id 4 doesn't have availability from 460 to 560",
+    )
+
+    const lengthUserAvsFullfilledAfter = await repo.getAvsFullfilledLength()
+
+    expect(lengthUserAvsFullfilledBefore).toEqual(lengthUserAvsFullfilledAfter)
+  })
+  it('Should raise error: professor already have classes on certain time into other schedule', async () => {
+    const repo = new ScheduleRepositoryMock()
+    const usecase = new UpdateAvailabilitiesFullfilledUsecase(repo)
+    const scheduleId = "2S-4CM-D5@2024(SCS)"
+    const avsFullfilledParam : AvailabilitiesFullfilledParam[] = [
+      {
+        userId:3,
+        possibilityId:"125e4567-e89b-12d3-a456-426614174001",
+        classId:"0a8c5357-1f07-5b24-9845-9318c47ac926"
+      },
+      {
+        userId: 3,
+        possibilityId: '113e4567-e89b-12d3-a456-426614174000',
+        classId: '0a8c5357-1f07-5b24-9845-9318c47ac924',
+      }
+    ]
+
+    const lengthUserAvsFullfilledBefore = await repo.getAvsFullfilledLength() 
+
+    await expect(usecase.execute(scheduleId, avsFullfilledParam)).rejects.toThrowError(
+      'The professor with id 3 is already assigned to schedule with id 2S-3CM-D5@2024(SCS)',
+    )
+
+    const lengthUserAvsFullfilledAfter = await repo.getAvsFullfilledLength()
+
+    expect(lengthUserAvsFullfilledBefore).toEqual(lengthUserAvsFullfilledAfter)
+  })
```

Comment:
EntityError tests here too

Reply:
I validate on controller, so the usecase waits for valid ids from controller